core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java

```diff
@@ -197,6 +197,11 @@ public long getTotalBytesSent() {
     }
 
     public boolean handleClientOperation(int operation, HttpRequestProcessorSelector selector, RescheduleContext rescheduleContext) {
+        if (IOOperation.HEARTBEAT == operation) {
```

Comment:
should be handled in the `switch` block, also if heartbeat is received I would assume this is not a "useful" run, no need to keep threads busy spining because of heartbeats

Reply:
fixed